### PR TITLE
Transfer Buffer service to use ingress library

### PR DESCRIPTION
**Pull request by knapii-developments**
Refactors Buffer's ingress, frontend config, backend config, and service charts to use the ingress library.

### CODE REVIEW

File: charts/buffer/templates/ingress/backendconfig.yaml [hunk 1]
```diff
@@ -1,22 +1,3 @@
...
-    healthyThreshold: 2
-    unhealthyThreshold: 2
-    type: HTTPS
-    port: 443
-    requestPath: /status
-  {{- if .Values.ingress.securityPolicy }}
-  securityPolicy:
-    name: {{ .Values.ingress.securityPolicy }}
-  {{- end }}
+{{- include "libchart.backendconfig" (list . "buffer.backendconfig" )  -}}
```
Comment (jack-r-warren) [inline charts/buffer/templates/ingress/backendconfig.yaml:1]: what happens here if ingress is disabled and the library chart isn't included? What defines `libchart.backendconfig`?

Comment (knapii-developments) [reply charts/buffer/templates/ingress/backendconfig.yaml:1]: So the file now looks like this:
```
{{- define "buffer.ingress" -}}
{{- end }}
```
and in the values file `ingress.enabled: false`?

Comment (jack-r-warren) [reply charts/buffer/templates/ingress/backendconfig.yaml:1]: Sorry, let me rephrase:
What happens if ingress is disabled? The ingress chart wouldn't exist as a dependency, so line 1 of
```
{{- include "libchart.backendconfig" (list . "buffer.backendconfig" ) -}}
{{- define "buffer.backendconfig" -}}
{{- end }}
```
would be referencing a "libchart.backendconfig" that doesn't exist. What happens then?

Comment (jack-r-warren) [reply charts/buffer/templates/ingress/backendconfig.yaml:1]: Almost--I think we should put the `include` inside the `if`, not before it, because really what we want to say is "if ingress is disabled, don't even try to access the `libchart.backendconfig` it defines"

Comment (knapii-developments) [reply charts/buffer/templates/ingress/backendconfig.yaml:1]: The same issue applies [here](https://github.com/broadinstitute/terra-helm/pull/506#discussion_r716972208).

Comment (jack-r-warren) [reply charts/buffer/templates/ingress/backendconfig.yaml:1]: Ah, maybe include works differently. I'd always used `template` which does support being inside a conditional https://github.com/broadinstitute/terra-helm/blob/85d1cf7b62ef30423896759c87ec6f5f66113dce/charts/rawls/templates/migration.yaml

**Review by jack-r-warren — APPROVED** [review]

**Review by jyang-broad — APPROVED** [review]
looks sane! Just as a academic exercise, if we were to say, push a 0.7.0 version of the chart, would all the dependent charts still be able to run 0.6.0 until they're ready (or maybe a 1.0.0 for major breaking changes kind of thing)

File: charts/buffer/templates/ingress/backendconfig.yaml (hunk 1 above)
Comment (knapii-developments) [reply charts/buffer/templates/ingress/backendconfig.yaml:1]: Yeah, they have different purposes. You can read more about it [here](https://helm.sh/docs/chart_template_guide/named_templates/#the-include-function). But in short: 

> Because template is an action, and not a function, there is no way to pass the output of a template call to other functions; the data is simply inserted inline.
